File: courses/old/csci1060_sp19/notes/22_cpp/cylinder.m
function area = cylinder(height, radius)
% function to compute the surface area of a cylinder
% Usage:  area = cylinder(height, radius)
% If not specified, radius is assumed to be 1
  if (nargin == 1)
    radius = 1;                 % default value for second parameter
  end
  base = pi .* radius.^2;
  area = 2 .* pi .* radius .* height + 2 .* base;    % the first result

File: courses/csci1060/notes/22_cpp/cylinder.cpp
#include <cmath>
#include <iostream>
using namespace std;

double cylinder(double height, double radius=1.0) {
  double base = M_PI * radius * radius;
  double area = 2 * M_PI * radius * height + 2 * base;
  return area;
}


int main() {
  double h, r;
  cout << "Height? ";
  cin >> h;
  cout << "Radius? ";
  cin >> r;
  double area = cylinder(h, r);
  cout << "Surface area is " << area << endl;
}
